src/common/inference/inference_input.rs

```diff
@@ -0,0 +1,108 @@
+use std::collections::HashMap;
+
+use api::rest::{Document, Image, InferenceObject};
+use serde::de::IntoDeserializer;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use storage::content_manager::errors::StorageError;
+
+use super::bm25::Bm25Config;
+use super::local_model::LocalModelConfig;
+use super::service::InferenceData;
+
+#[derive(Debug, Serialize, Clone)]
+pub struct InferenceInput {
+    pub data: Value,
+    pub data_type: String,
+    pub model: String,
+    pub options: Option<HashMap<String, Value>>,
+}
+
+const LOCAL_MODEL_KEY: &str = "local_model";
+const BM_25_MODEL: &str = "bm25";
+
+impl InferenceInput {
+    /// Tries to parse the given options as `Bm25Config`. Returns an error if the options can't be parsed.
+    fn try_parse_bm25_config(options: &HashMap<String, Value>) -> Result<Bm25Config, StorageError> {
+        Bm25Config::deserialize(options.clone().into_deserializer())
+            .map_err(|err| StorageError::bad_input(format!("Invalid BM25 config: {err:#?}")))
+    }
+
+    /// Attempts to parse the input's options into a local model config.
+    pub fn try_parse_local_model_input(&self) -> Result<Option<LocalModelConfig>, StorageError> {
+        let Some(options) = self.options.as_ref() else {
+            return Ok(None);
+        };
+
+        let Some(local_model_field) = options.get(LOCAL_MODEL_KEY) else {
```

Comment:
I still don't understand why this is needed

Reply:
If we add a different local 'model', next to bm25, which also has tokenizer configs and a user only specifies those (so no Bm25 specific parameters), we can't distinguish which model should be used.
This requires the user to specify which local model to use.

We could use some dedicated model name e.g. 'local/bm25' to specify which local model to use. However I thought hard-coding some specific model name to be potentially confusing, especially if something like 'local/bm25' works but 'bm25' throws an error about some 'address' not being found. If you prefer this approach though, I'm totallyfine with changing it.